tsdoc/src/parser/TokenReader.ts

```diff
@@ -0,0 +1,126 @@
+import { Token, TokenKind } from './Token';
+import { TokenRange } from './TokenRange';
+import { ParserContext } from './ParserContext';
+
+export class TokenReader {
+  public readonly tokens: ReadonlyArray<Token>;
+
+  private readonly _parserContext: ParserContext;
+  private _rangeStartIndex: number;
+  private _currentIndex: number;
+
+  public constructor(parserContext: ParserContext) {
+    this._parserContext = parserContext;
+    this.tokens = parserContext.tokens;
+    this._rangeStartIndex = 0;
+    this._currentIndex = 0;
+  }
+
+  public extractQueue(): TokenRange {
```

Comment:
>Queue [](start = 16, length = 5)

Field?

Reply:
Sequence?

---
In reply to: [209787204](https://github.com/Microsoft/tsdoc/pull/39#discussion_r209787204) [](ancestors = 209787204)

Reply:
I renamed the class to `TokenSequence`.  And I renamed `extractQueue()` to `extractAccumulatedSequence()`

---
In reply to: [209787255](https://github.com/Microsoft/tsdoc/pull/39#discussion_r209787255) [](ancestors = 209787255,209787204)